modules/react/text-input/lib/InputGroup.tsx

```diff
@@ -42,7 +55,69 @@ export const InputGroupInnerEnd = createComponent('div')({
 
 export const InputGroupInput = createComponent(TextInput)({
   Component(elemProps: ExtractProps<typeof Flex>, ref, Element) {
-    return <Flex ref={ref} as={Element} width="100%" {...elemProps} />;
+    const inputRef = React.useContext(InputGroupContext);
+    const elementRef = useForkRef(ref, inputRef);
+    return <Flex ref={elementRef} as={Element} width="100%" {...elemProps} />;
+  },
+});
+
+export interface ClearInputButtonProps {}
+
+/**
+ * A clear input button. This can be a component later.
+ */
+export const ClearInputButton = createComponent(TertiaryButton)({
+  Component(elemProps: ClearInputButtonProps, ref, Element) {
```

Comment:
instead of defining probs can we have something like `ExtractProps<typeof TertiaryButton>` ?

Reply:
Good point